src/core_plugins/kibana/public/management/react/index.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render } from 'react-dom';
+import { Provider } from 'react-redux';
+import { globals } from './globals';
+import { store } from './store';
+
+import { App } from './app/index';
+import {
+  IndexPatternCreate,
+  IndexPatternList,
+  IndexPatternView,
+} from './index-pattern';
+
+import {
+  fetchIndexPatterns,
+  initData,
+} from 'plugins/kibana/management/react/store/actions';
+
+import {
+  getIndexPatterns,
+} from 'plugins/kibana/management/react/store/reducers';
+
+const renderPage = (page, targetId) => {
+  render(
+    <Provider store={store}>
+      <App>
+        {page}
+      </App>
+    </Provider>,
+    document.getElementById(targetId)
+  );
+};
+
+const ReactApp = {
+  init: ($injector, managementConfig) => {
+    ReactApp.setGlobals($injector);
+    const kbnVersion = $injector.get('kbnVersion');
+    store.dispatch(initData(kbnVersion, managementConfig));
+  },
+
+  setGlobals: $injector => {
+    globals.es = $injector.get('es');
+    globals.indexPatterns = $injector.get('indexPatterns');
+    globals.config = $injector.get('config');
+    globals.kbnUrl = $injector.get('kbnUrl');
```

Comment:
So far I've had success using react hashrouter and history.push instead of kbnUrl. Not saying that's how it should be, but we differ in our approaches in this aspect.

I do really like the `globals` as a way to expose angular portions.  I'm doing something similar with the notifier to expose errors (though rather than expose the whole Notifier class, I'm just exposing an onNotifyError function) - https://github.com/elastic/x-pack-kibana/commit/16138ccbd0e84551f56fc920ad665d5022185501#diff-092d60ff46d2b710c92fd6958b923decR20

Reply:
Btw, instead of importing `globals`, maybe rely on https://github.com/gaearon/redux-thunk#injecting-a-custom-argument to get them into actions? That makes it a bit more obvious that they can't be used in other places.

Reply:
Tangential to the discussion so far, I think ideally we'd want to refactor these Angular services to be light wrappers around a pure-JS module. This way Angular code can continue to depend upon the original Angular service, but React code can import the pure-JS module, thus reducing our React code's coupling with Angular.

Reply:
++ totally agree with this @cjcenizal - but also think it will end up being a very big endeavor in and of itself.  Like a year ago I wanted to explore de-angularizing some of our event emitter code but it is soooo tightly coupled to angular.  And with the way we pass angular promises around...  and use Private to inject angular stuff eeeverywhere...